abm_sketch.py

```diff
@@ -157,13 +164,13 @@ def initiate_task(self, fixed_pair=None):
         start, end = [-1, -1] if not fixed_pair else fixed_pair
 
         if fixed_pair and start == end:
-            print("changing fixed pair: they're the same!")
+            self.log("changing fixed pair: they're the same!")
```

Comment:
I think should just raise an error here -- fixed pair shouldn't be changed, right? do some callers need this behavior?